cupy/_core/core.pyx

```diff
@@ -278,16 +278,18 @@ cdef class _ndarray_base:
         if runtime._is_hip_environment:
             raise AttributeError(
                 'HIP/ROCm does not support cuda array interface')
+
+        str, descr = _dtype.get_str_and_descr(self.dtype)
         cdef dict desc = {
             'shape': self.shape,
-            'typestr': self.dtype.str,
-            'descr': self.dtype.descr,
+            'typestr': str,
+            'descr': descr,
         }
         cdef int ver = _util.CUDA_ARRAY_INTERFACE_EXPORT_VERSION
         cdef intptr_t stream_ptr
 
         if ver == 3:
-            stream_ptr = stream_module.get_current_stream_ptr()
+            stream_ptr = stream_module.get_current_stream().ptr
```

Comment:
Not sure this is good:  It seems faster(?!) and is what dlpack uses so I kept it, but I am not sure about the distinction at all here.

Reply:
Hmm... It looks identical.
https://github.com/cupy/cupy/blob/397fd017e235d0db153655a46ea0fd2f28f8e445/cupy/cuda/stream.pyx#L63-L67
https://github.com/cupy/cupy/blob/397fd017e235d0db153655a46ea0fd2f28f8e445/cupy_backends/cuda/stream.pyx#L34-L43